### PR TITLE
[backup] Fix on demand backup

### PR DESCRIPTION
**Pull request by hiaga**
**Description**

fixing on demand backups for SQL workloads - modifying when a user should or shouldn't give the value for retain-until

**Testing Guide**

az backup protection backup-now -v vaultName -g rgName--backup-management-type AzureWorkload --workload-type MSSQL -c "containerName" -i "itemName" --backup-type "CopyOnlyFull/Full" --retain-until 06-04-2021  --debug

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by hiaga** [comment]
@yungezz , @fengzhou-msft, @houk-ms : can you please review this PR to fix the on demand backup for SQL and HANA workloads.

**Review by yungezz — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ __pycache__/
 
 # Virtual environment
 env/
+env1/
```
Comment (qwordy) [inline .gitignore:7]:
```suggestion
```

**Review by qwordy — APPROVED** [review]

**Comment by hiaga** [comment]
@qwordy : I have made the change as suggested, pls approve and merge.

**Comment by qwordy** [comment]
> 
> 
> @qwordy : I have made the change as suggested, pls review and merge.

@fengzhou-msft Code owner review required. Please review and merge.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**